librz/core/cbin.c

```diff
@@ -4484,3 +4484,81 @@ RZ_API char *rz_core_bin_method_flags_str(ut64 flags, int mode) {
 out:
 	return rz_strbuf_drain(buf);
 }
+
+static RzCoreStringKind get_string_kind(const ut8 *buf, ut64 len) {
+	rz_return_val_if_fail(buf && len, RZ_CORE_STRING_KIND_UNKNOWN);
+	ut64 needle = 0;
+	int rc, i;
```

Comment:
i and rc should be ut64

Reply:
Why? `rz_utf8_decode()` returns `int`, sometimes `-1` actually.

Reply:
because you are comparing unsigned int with signed int of different sizes `needle + rc + 2 < len`